dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/curator/CuratorZookeeperClient.java

```diff
@@ -109,6 +110,14 @@ public void createEphemeral(String path) {
         try {
             client.create().withMode(CreateMode.EPHEMERAL).forPath(path);
         } catch (NodeExistsException e) {
+            try {
+                if (!checkEphemeralBelongToCurrentSession(path)) {
+                    deletePath(path);
+                    createEphemeral(path);
+                }
+            }catch (Exception e1) {
+                throw new IllegalStateException(e.getMessage(), e1);
+            }
         } catch (Exception e) {

```

Comment:
I find this method will get triggered whenever ConnectionState.RECONNECT happens, which can be caused on the root by either connection recover or new session creation. I think it would be better if we can recognize this different condition by checking session id at the entry point of the callback.

Reply:
I agree with you that it is better to do so.